integration-tests/ccip-tests/actions/ccip_helpers.go

```diff
@@ -1888,26 +1889,28 @@ func (sourceCCIP *SourceCCIPModule) CCIPMsg(
 func (sourceCCIP *SourceCCIPModule) SendRequest(
 	receiver common.Address,
 	gasLimit *big.Int,
-) (common.Hash, time.Duration, *big.Int, error) {
+) (common.Hash, time.Duration, *big.Int, []byte, error) {
 	var d time.Duration
 	destChainSelector, err := chainselectors.SelectorFromChainId(sourceCCIP.DestinationChainId)
 	if err != nil {
-		return common.Hash{}, d, nil, fmt.Errorf("failed getting the chain selector: %w", err)
+		return common.Hash{}, d, nil, nil, fmt.Errorf("failed getting the chain selector: %w", err)
 	}
 	// form the message for transfer
 	msg, err := sourceCCIP.CCIPMsg(receiver, sourceCCIP.Common.AllowOutOfOrder, gasLimit)
 	if err != nil {
-		return common.Hash{}, d, nil, fmt.Errorf("failed forming the ccip msg: %w", err)
+		return common.Hash{}, d, nil, nil, fmt.Errorf("failed forming the ccip msg: %w", err)
 	}
 
+	msgData := msg.Data
+
 	fee, err := sourceCCIP.Common.Router.GetFee(destChainSelector, msg)
 	if err != nil {
 		log.Info().Interface("Msg", msg).Msg("CCIP msg")
 		reason, _ := blockchain.RPCErrorFromError(err)
 		if reason != "" {
-			return common.Hash{}, d, nil, fmt.Errorf("failed getting the fee: %s", reason)
+			return common.Hash{}, d, nil, nil, fmt.Errorf("failed getting the fee: %s", reason)
```

Comment:
nit: As we have msgData, we can send that instead of nil from here onwards even if it is failure.
```suggestion
			return common.Hash{}, d, nil, msgData, fmt.Errorf("failed getting the fee: %s", reason)
```

Reply:
Done